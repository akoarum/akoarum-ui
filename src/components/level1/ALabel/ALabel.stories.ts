import { storiesOf } from '@storybook/vue'
import { withKnobs, text, boolean } from '@storybook/addon-knobs'
import { withInfo } from 'storybook-addon-vue-info'
import ALabel from './ALabel.vue'

storiesOf('level1/ALabel', module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs)
  .add('Normal', () => ({
    components: { ALabel },
    props: {
      label: { default: text('テキスト', 'ラベル') },
      span: { default: boolean('span要素', false) }
    },
    template: `
      <ALabel :span="span">{{ label }}</ALabel>
    `
  }), { info: {} })
  .add('Error', () => ({
    components: { ALabel },
    props: {
      label: { default: text('テキスト', 'ラベル') },
      span: { default: boolean('span要素', false) }
    },
    template: `
      <ALabel :span="span" error>{{ label }}</ALabel>
    `
  }), { info: {} })

