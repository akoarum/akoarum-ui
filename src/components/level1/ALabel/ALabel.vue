<script lang="ts">
import { CreateElement, VNode } from 'vue'
import { defineComponent, computed } from '@vue/composition-api'

interface Props {
  error: boolean
  span: boolean
}

export default defineComponent({
  props: {
    error: { type: Boolean },
    span: { type: Boolean }
  },
  setup(props: Props) {
    const styledClasses = computed<string[]>(() => {
      const classes: string[] = ['aLabel']
      if (props.error) classes.push('-error')
      return classes
    })
    return { styledClasses }
  },
  render(createElement: CreateElement): VNode {
    return createElement(
      this.span ? 'span' : 'label',
      {
        class: this.styledClasses
      },
      this.$slots.default
    )
  }
})
</script>

<style lang="scss" scoped>
.aLabel {
  font-size: 14px;
  font-weight: bold;

  &.-error {
    color: $color-caution;
  }
}
</style>
