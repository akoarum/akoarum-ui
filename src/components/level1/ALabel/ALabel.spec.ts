import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueCompositionApi from '@vue/composition-api'
import ALabel from './ALabel.vue'

describe('ALabel', () => {
  const localVue = createLocalVue()
  let wrapper

  localVue.use(VueCompositionApi)

  beforeEach(() => {
    wrapper = shallowMount(ALabel, {
      propsData: {},
      localVue
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    it('styledClasses: propsから見た目を制御できる', () => {
      wrapper.setProps({ error: true })
      expect(wrapper.vm.styledClasses).toEqual(['aLabel', '-error'])
    })
  })
})
