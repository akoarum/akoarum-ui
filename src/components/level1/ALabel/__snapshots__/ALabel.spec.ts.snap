// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ALabel initialized snapshot 1`] = `
<label
  class="aLabel"
/>
`;
